### ISSUE TITLE
ultisnips causing slowness displaying text in insert mode

### ISSUE THREAD

**Issue by rafaelleru** [SirVer/ultisnips#1496]
neovim has been displaying text while writting for a couple of days, after upgrading it to the latest version and trying multiple versions I tried to profile neovim to see what happened. Ultisnips function UltiSnips#TrackChange() is taking near 3 seconds to run, causing a big delay in text rendering in insert mode.

**Expected behavior:**
vim to be as fast as always

**Actual behavior:**
vim is too slow to render text in insert mode

**Steps to reproduce**
load teh extension
write text on any file

-----

- **Operating System**:  Ubuntu 22.04
- **Vim Version**: neovim 0.8.0
- **UltiSnips Version**: latest
- **Python inside Vim**:  3.10.6
- **Docker repo/vimrc**:  https://github.com/rafaelleru/dotfiles/tree/ansible/config/nvim

# Output of the profile file for the given function:

```
FUNCTION  UltiSnips#TrackChange()
    Defined: ~/.vim/plugged/ultisnips/autoload/UltiSnips.vim:176
Called 56 times
Total time:   2.833268
 Self time:   2.833268

count  total (s)   self (s)
   56              2.833192     py3 UltiSnips_Manager._track_change()

FUNCTION  provider#python3#Call()
    Defined: /usr/local/share/nvim/runtime/autoload/provider/python3.vim:26
Called 56 times
Total time:   2.832244
 Self time:   2.832244

count  total (s)   self (s)
   56              0.000319   if s:err != ''
                                return
   56              0.000070   endif
   56              0.000257   if !exists('s:host')
                                let s:rpcrequest = function('rpcrequest')
                            
                                " Ensure that we can load the Python3 host before bootstrapping
                                try
                                  let s:host = remote#host#Require('legacy-python3-provider')
                                catch
                                  let s:err = v:exception
                                  echohl WarningMsg
                                  echomsg v:exception
                                  echohl None
                                  return
                                endtry
   56              0.000029   endif
   56              2.830506   return call(s:rpcrequest, insert(insert(a:args, 'python_'.a:method), s:host))


```

* H

**Comment by kuator** [comment]
neovim is not supported

**Comment by SirVer** [comment]
Fixed in #1552
